### ISSUE TITLE
Regression in 2.3.x regarding config path configuration

### ISSUE THREAD

**Issue by aisven** [acederberg/pydantic-settings-yaml#29]
Last working version: 2.2.0

First broken version: 2.3.0

Regarding this issue, I also commented on the past pull request that contains the code change that triggered this issue.

https://github.com/acederberg/pydantic-settings-yaml/pull/25

So basically, the newly introduced validation seems too strict. It seems to disallow usages that previously worked.

Here is how my config looks like, it used to work:

```
    model_config = YamlSettingsConfigDict(
        yaml_files={
            "config/config.yaml": YamlFileConfigDict(
                subpath="asyncio_settings",
                required=True,
            )
        }
    )
```

This is somehow not accepted by the validation. Specifically, it raises the `ValueError` with the message starting with `All items in files must be strings. The following are`.

@acederberg 

@heliocastro

**Comment by aisven** [comment]
@acederberg @heliocastro Here is also a code pointer to where that ValueError is, unexpectedly, raised.

https://github.com/acederberg/pydantic-settings-yaml/blob/feature/path-from-environment/yaml_settings_pydantic/__init__.py#L242

At least as of today it points to the correct line of code. I allow myself to also copy the code around that line to this comment:

```

        keys_invalid = set(item for item in value if not isinstance(item, str))
        if len(keys_invalid):
            raise ValueError(
                "All items in `files` must be strings. The following are "
```

**Comment by acederberg** [comment]
Hi @aisven!

Thank you for bringing this to my attention. From what I can tell, this is (in your case) broken because all of these ``values`` should have type ``Path``. In the current ``release/v2`` branch this is correct, see [the code on that branch](https://github.com/acederberg/pydantic-settings-yaml/blob/release/v2/yaml_settings_pydantic/__init__.py#L261). Also, in the branch I'm currently working on I have

```python

    keys_invalid = {item for item in values if not isinstance(item, Path)}
    if len(keys_invalid):
        raise ValueError(
            "All items in `files` must be strings. The following are "
            f"not strings: `{keys_invalid}`."
        )
```

which I branched after tag ``2.3.1``. Further, if I

```sh
git checkout 2.3.0
```

this chunk of code does not appear to be as stated. The ``isinstance`` check should be checking for ``Path`` in the latest releases (``2.3.x``). 

~~Could you confirm your version? This is best done by running~~

```python
python -m yaml_settings_pydantic version
```

**Comment by acederberg** [comment]
@aisven 

I also installed this into a ``venv`` and looked directly at the source and this line does not appear. This was done like

```sh
source .venv
poetry add yaml_settings_pydantic
vim .venv/lib/python3.12/site-packages/yaml_settings_pydantic/__init__.py
```

I have figured out what is wrong and will address it as soon as possible. For now, do

~~~python
    model_config = YamlSettingsConfigDict(
        yaml_files={
            Path("config/config.yaml"): YamlFileConfigDict(
                subpath="asyncio_settings",
                required=True,
            )
        }
    )
~~~

**Comment by acederberg** [comment]
Fixed in release [2.3.2](https://pypi.org/manage/project/yaml-settings-pydantic/releases/).
